### ISSUE TITLE
`createReadStream` crashes with `TypeError: Cannot read property 'bind' of undefined`

### ISSUE THREAD

**Issue by sunrize531** [Level/levelup#613]
After updating to the version 3.1.1 method `createReadStream`  breaks with the exception when trying to bind `destroy` method on `Readable` stream. 

```
TypeError: Cannot read property 'bind' of undefined
    at new ReadStream (xxx\node_modules\level-iterator-stream\index.js:16:31)
    at LevelUP.readStream.LevelUP.createReadStream (xxx\node_modules\levelup\lib\levelup.js:274:10)
```

I think there might be a bug or change of behaviour in `readable-stream` module (it's installed version 2.2.7 for me), cause it won't use a native nodejs Stream unless `READABLE_STREAM` environment variable is explicitly set to `"disable"`. 

Also, this call wouldn't work on node 6.x, cause `destroy` method was only added in nodejs 8.0. Might be ok if you don't maintain nodejs 6.x compatibility anymore, otherwise worth handling it more robustly.

Running nodejs 8.9.1.

**Comment by vweevers** [comment]
This should have been fixed by https://github.com/Level/levelup/pull/596.

Can you share a stack trace?

**Comment by vweevers** [comment]
Hm, the `destroy` method should be present in `readable-stream` 2.x:

https://github.com/nodejs/readable-stream/blob/v2.3.6/lib/_stream_readable.js#L217

@sunrize531 did you set `READABLE_STREAM=disable`?

> Also, this call wouldn't work on node 6.x, cause destroy method was only added in nodejs 8.0. Might be ok if you don't maintain nodejs 6.x compatibility anymore, otherwise might worse handling this more robustly.

The point of `readable-stream` is that it doesn't matter which node version you're using, you always get the same streams implementation.

**Comment by sunrize531** [comment]
No I didn't set environment variable.
Also I'm fairly sure it was using `readable-stream` implementation on my machine, and i couldn't see `destroy` in it's prototype or anywhere else in the debugger.

**Comment by vweevers** [comment]
What's the output of `npm ls level-iterator-stream readable-stream`?

**Comment by sunrize531** [comment]
I've already reverted it, let me try to reproduce it when I'm done with what I'm doing now. 
Will update you later today.

**Comment by sunrize531** [comment]
```
+-- leveldown@4.0.1
| `-- prebuild-install@4.0.0
|   +-- UNMET DEPENDENCY npmlog@4.1.2
|   | `-- are-we-there-yet@1.1.5
|   |   `-- readable-stream@2.2.7  deduped
|   `-- tar-fs@1.16.3
|     `-- tar-stream@1.6.2
|       +-- bl@1.2.2
|       | `-- readable-stream@2.3.6
|       `-- readable-stream@2.3.6
+-- levelup@3.1.1
| `-- level-iterator-stream@3.0.0
|   `-- readable-stream@2.2.7  deduped
`-- superagent@3.8.2
  `-- readable-stream@2.2.7
```

**Comment by vweevers** [comment]
Ah. That version of `readable-stream` doesn't have `destroy`: https://github.com/nodejs/readable-stream/blob/v2.2.7/lib/_stream_readable.js

We need to bump that dependency in `level-iterator-stream`, so that it doesn't get deduped by npm with a version that's too old.

**Comment by sunrize531** [comment]
Yeah, looks like really easy to fix, thanks!

**Comment by vweevers** [comment]
Fixed in `level-iterator-stream@3.0.1`. Thanks for reporting!
